flink-ml-iteration/src/main/java/org/apache/flink/ml/iteration/EpochWatermarkAware.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.iteration;
+
+import java.util.function.Supplier;
+
+/**
+ * Operators or UDF implements this method would be provided with an supplier that provides the
+ * current rounds of the current element.
+ */
+public interface EpochWatermarkAware {
+
+    void setEpochWatermarkSupplier(Supplier<Integer> epochWatermarkSupplier);
```

Comment:
`ProgressTrackerListener.onEpochWatermarkIncrement` provides operators with the information of latest epoch watermark. Why do we still need this class and method to provide the same information?

Reply:
Sorry this class should be `EpochAware`. It is used to provide the epoch of current record.